src/sciline/pipeline.py

```diff
@@ -750,3 +760,42 @@ def bind_and_call(
         if not return_tuple:
             return results[0]
         return results
+
+    def _repr_html_(self) -> str:
+        providers_without_parameters = (
+            (p, tuple(), v) for p, v in self._providers.items()
+        )  # type: ignore[var-annotated]
+        providers_with_parameters = (
+            (p, c, v)
+            for p in self._subproviders
+            for c, v in self._subproviders[p].items()
+        )
+        providers = groupby(
+            lambda p: p.kind,
+            (
+                ProviderDisplayData(
+                    origin,
+                    args,
+                    k := _kind_of_provider(value),
+                    value() if k != 'function' else value,
+                )
+                for origin, args, value in chain(
+                    providers_with_parameters, providers_without_parameters
+                )
+            ),
+        )
```

Comment:
Will this make one entry in the repr **per row** (and column) of a param table? Or am I misreading this?

Reply:
You're misreading it. Take a look at the examples in the screenshots I've posted in this PR. Parameter Tables use their own built in html repr.

However, it is quite messy to reconstruct the parameter tables from the `_provider` and `_subprovider` dicts, so it would be good if we could do that in a better way.

Reply:
I thought we had decided to combine everything into a single table? I had thought those screenshots were thus outdated.

Reply:
Okay I see. The parameters were merged with the providers in a single table. But the parameter tables are still displayed separately. Do you think it'd be better to merge them in the larger table as well? I think that it might be more messy. The parameter tables are self contained and have a html repr already so I thought it sensible to use that and display them separately.

Reply:
Imho a user can display the param table themselves if they want to. The HTML repr *of the pipeline should* give info *about the pipeline* (i.e., mainly the structure), and as such I would consider each param table "column" as a single entity. That is, simply display a single summary line in the repr, instead of all the values. After all there might be thousands of rows in a param table?

Reply:
That makes a lot of sense. I'll do something like that.

Reply:
![Screenshot from 2023-12-20 14-23-42](https://github.com/scipp/sciline/assets/20954731/8004318b-53d6-43a9-bea4-acb2d8b532b1)

Reply:
I decided to put values of parameters in the table, but below a `<details>` tag. This way they don't take up space when they are not expanded, but can be quickly expanded and displayed using their own `_repr_html_`.

Reply:
![Screenshot from 2023-12-20 14-27-57](https://github.com/scipp/sciline/assets/20954731/b21ca26d-47cf-43e1-b743-891068d6ea0a)

Reply:
Can we swap the "Source" and "Value" columns

Reply:
As I said before, I would be worried about size and lag when using this with large param tables. I think we should not include full nested HTML reprs of each value. Listing the size should be a good start.

It is generally always easier to make things more complicated later, than to remove it (because then some users who learned to love and rely on a feature will become upset).

Reply:
I agree it might be a problem, but I think it is a unusual problem and one that is difficult to protect against in general. Displaying the html repr of any DataArray might lag if it is large enough. I also think being able to view the values of all parameters in the workflow like this is quite useful.

We talked before about displaying a reduced view of variables like is done in datagroup. The issue with that approach in this case is that we don't want the sciline code to contain lots of logic about how to display scipp datatypes.

I don't know what you mean by listing the size, values of parameters can have any type, they have no concept of size.

Reply:
> I'm not sure about what you mean by listing the size, values of parameters can have any type and they don't have a common concept of size.

I meant the number of rows.

Reply:
Yes it switched a bit back and forth and I understand it got confusing.

When we talked at my desk the other day we decided to only display the length of parameter tables. That's what's done now.

Now I thought we were talking about *parameters* that are not in tables. Just regular input values like `QBins` or `Filename[Sample`. Those are currently displayed below a details tag in case they have a html repr. Otherwise they're just stringified.

> Shouldn't the param-table info go into the source column, not the value column? Something like ParamTable(int, length=1000) maybe?

That sounds like a good idea. Maybe the source column should change name to "info" or something like that instead.